src/sidebar/components/svg-icon.js

```diff
@@ -16,17 +23,30 @@ const icons = {
  * This matches the way we do icons on the website, see
  * https://github.com/hypothesis/h/pull/3675
  */
-function SvgIcon({ name }) {
+function SvgIcon({ name, size }) {
   if (!icons[name]) {
     throw new Error(`Unknown icon ${name}`);
   }
   const markup = { __html: icons[name] };
-  return <span dangerouslySetInnerHTML={markup} />;
+
+  const element = useRef();
+  useLayoutEffect(() => {
+    const svg = element.current.querySelector('svg');
+    if (typeof size === 'number') {
+      svg.style.width = `${size}px`;
+      svg.style.height = `${size}px`;
+    }
+  }, [element, size]);
```

Comment:
There are two types of effects in React, "regular" effects (`useEffect`) and "layout" effects. The difference AIUI is that "layout" effects run _before_ the screen repaints and regular effects run after.